### ISSUE TITLE
Support setting OwnerReferences on OB,Secret,CM to handle deletion when deleting the provisioner

### ISSUE THREAD

**Issue by guymguym** [kube-object-storage/lib-bucket-provisioner#172]

**Comment by guymguym** [comment]
We want to handle the case where we are uninstalling the provisioner while OBC/OBs already exist for applications. 

I think the desired behavior is that existing OBs will get deleted and the OBCs will change their status to represent a no-provisioner phase.

**Comment by jeffvance** [comment]
> I think the desired behavior is that existing OBs will get deleted and the OBCs will change their status to represent a no-provisioner phase.

This seems reasonable.

**Comment by jeffvance** [comment]
I propose a change to the title since OBs, Secrets and CMs already have ownerReference set.

**Comment by copejon** [comment]
The OBs are a little trickier.  [Cluster-scoped dependents can only specify cluster-scoped owners, but not namespace-scoped owners.](https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/#controlling-how-the-garbage-collector-deletes-dependents).

**Comment by jeffvance** [comment]
> The OBs are a little trickier.

Good point! I forget ownerRefs cannot be applied to non-namespaced resources.
